src/libexpr/primops.cc

```diff
@@ -1836,6 +1836,39 @@ static RegisterPrimOp primop_readDir({
     .fun = prim_readDir,
 });
 
+/* Extend single element string context with another output. */
+static void prim_outputOf(EvalState & state, const PosIdx pos, Value * * args, Value & v)
+{
+    SingleDerivedPath drvPath = state.coerceToSingleDerivedPath(pos, *args[0], "while evaluating the first argument to builtins.outputOf");
+
+    std::string_view outputName = state.forceStringNoCtx(*args[1], pos, "while evaluating teh second argument to builtins.outputOf");
+
+    NixStringContextElem::Built path2 {
+        .drvPath = make_ref<SingleDerivedPath>(drvPath),
+        .output = std::string { outputName },
+    };
+
+    v.mkString(
+        DownstreamPlaceholder::fromSingleDerivedPathBuilt(path2).render(),
+        { path2 });
+}
+
+static RegisterPrimOp primop_outputOf({
+    .name = "__outputOf",
+    .args = {"drv path", "output name"},
+    .doc = R"(
+      Return path (actually placeholder path) to the output of the given drv.
+      The drv path may itself be a placeholder, which permits chaining this primop.
+
+      For instance, `builtins.outputOf (builtins.outputOf myDrv "out) "out"`
+      will return a placeholder for the output of the output of `myDrv`,
+      interpreted as a derivation.
+
+      It may help to compare to the `->` operator in C, which can also by
```

Comment:
I removed that and referenced `^` in the CLI syntax for derived paths, which is what it actually corresponds to.

I don't think `.` is so good because the field in some sense "already exists", attributes in the attribute set being strict and all. but talking about `^` side-steps that issue.